### DISCUSSION TITLE
403 error after setting up data-importer

### DISCUSSION THREAD

**Discussion by filip-van-hoeckel** [firefly-iii/firefly-iii#9518] [Firefly III]
FF latest, NGINX on Ubuntu 20. FF is running happily. I installed the importer as per documentation. I use virtual hosts, so my importer would be at http://ff-importer.test/. When opening I get a 403 error. I checked folder rights and they do look correct to me. This is my host config 

`server {
    listen 80;
    listen [::]:80;

    server_name  ff-importer.test www.ff-importer.test;
    root   /var/www/data-importer/public;
    index  index.php;

    access_log /var/log/nginx/ff-importer.test.access.log;
    error_log /var/log/nginx/ff-importer.test.error.log;

    client_max_body_size 100M;

    autoindex off;

    location / {
    try_files $uri $uri/ /index.php$is_args$args;
      }

    location ~ \.php$ {
         include snippets/fastcgi-php.conf;
         fastcgi_pass unix:/var/run/php/php8.3-fpm.sock;
         include fastcgi_params;
         fastcgi_intercept_errors on;
    }
}`

In the error log I see 
`2024/12/03 20:59:20 [error] 74340#74340: *574 directory index of "/var/www/data-importer/" is forbidden,...` 
which strikes me as odd as rights should be correct as I set these as per documentation. Please advise, thanks.

**Comment by JC5** [comment]
Try this config:

```
     try_files $uri $uri/ /index.php?$query_string;
```

**Comment by filip-van-hoeckel** [comment]
After a nginx restart I can see the interface, however the css 404s.

**Reply by filip-van-hoeckel** [reply]
As a matter of fact I can see it's trying to load assets from "/data-importer/build/assets/app-amkGEgGS.css". I'd expect it to load from the root of the installation (root = /build/) assets folder would be "/build/assets/".

**Reply by JC5** [reply]
I would expect that as well. What does the html source of the page say?

**Reply by filip-van-hoeckel** [reply]
`<link rel="preload" as="style" href="[/data-importer/build/assets/app-amkGEgGS.css](http://ff-importer.test/data-importer/build/assets/app-amkGEgGS.css)" /><link rel="stylesheet" href="[/data-importer/build/assets/app-amkGEgGS.css](http://ff-importer.test/data-importer/build/assets/app-amkGEgGS.css)" />` for styling and 
`<link rel="modulepreload" href="[/data-importer/build/assets/index-CXBe0dho.js](http://ff-importer.test/data-importer/build/assets/index-CXBe0dho.js)" /><link rel="modulepreload" href="[/data-importer/build/assets/bootstrap-CJTxePce.js](http://ff-importer.test/data-importer/build/assets/bootstrap-CJTxePce.js)" /><link rel="modulepreload" href="[/data-importer/build/assets/vendor-DCxVLCSx.js](http://ff-importer.test/data-importer/build/assets/vendor-DCxVLCSx.js)" /><script type="module" src="[/data-importer/build/assets/index-CXBe0dho.js](http://ff-importer.test/data-importer/build/assets/index-CXBe0dho.js)"></script>
` for JS.

**Reply by JC5** [reply]
Mmmm. The prefix is very strange. Did you set the ASSET_URL by any chance?

**Reply by filip-van-hoeckel** [reply]
I did. So I removed it. Now it renders correctly but it throws an error `Error accessing "http://192.168.178.11/api/v1/about?". Status code is 404. Body is: <html> <head><title>404 Not Found</title></head> <body> <center><h1>404 Not Found</h1></center> <hr><center>nginx/1.26.2</center> </body> </html>`. However I set 'FIREFLY_III_URL=http://firefly.test' and left vanity url empty. So how did it get to the IP-address in the first place?

**Reply by filip-van-hoeckel** [reply]
OK, so I removed all URLs from the .env then got a normal interface (!) So I input Client ID after generating it, and the virtualhost for the URL (including protocol). That gives a curl error 6 after authorising the connection. If I set the url to the IP-adress then the whole process fails with a 404. Out of options for now....

**Reply by filip-van-hoeckel** [reply]
OK, so rather than setting the url in importer config, one should add the virtualhost to the hosts file on the server.
